pkg/instancetype/instancetype.go

```diff
@@ -851,6 +853,37 @@ func (m *InstancetypeMethods) inferDefaultsFromDataVolumeSourceRef(sourceRef *v1
 	return "", "", fmt.Errorf("unable to infer defaults from DataVolumeSourceRef as Kind %s is not supported", sourceRef.Kind)
 }
 
+func applyInstanceTypeAnnotations(annotations map[string]string, target metav1.Object) (conflicts Conflicts) {
+	if target.GetAnnotations() == nil {
+		target.SetAnnotations(make(map[string]string))
+	}
+
+	existingAnnotations := target.GetAnnotations()
+	for key, value := range annotations {
+		if val, exists := existingAnnotations[key]; exists && val != value {
+			conflicts = append(conflicts, k8sfield.NewPath("annotations", key))
+			continue
+		}
+		target.GetAnnotations()[key] = value
```

Comment:
Can you reuse `existingAnnotations` from L861 here? Maybe rename it to `targetAnnotations`?